core/client/src/backend.rs

```diff
@@ -105,6 +124,45 @@ pub trait BlockImportOperation<Block, H> where
 	fn mark_head(&mut self, id: BlockId<Block>) -> error::Result<()>;
 }
 
+/// Finalize Facilities
+pub trait Finalizer<Block: BlockT, H: Hasher<Out=Block::Hash>, B: Backend<Block, H>> {
+	/// Mark all blocks up to given as finalized in operation. If a
+	/// justification is provided it is stored with the given finalized
+	/// block (any other finalized blocks are left unjustified).
+	///
+	/// If the block being finalized is on a different fork from the current
+	/// best block the finalized block is set as best, this might be slightly
+	/// innacurate (i.e. outdated), usages that require determining an accurate
+	/// best block should use `SelectChain` instead of the client.
+	fn apply_finality(
```

Comment:
So the only difference between both functions is the `operation` parameter? Can we not just make the `operation` an `Option`?

Reply:
well, there is a significant difference between the two functions that I believe would be to implicit with just an `option` to notice: one applies the finality to an uncommitted operation - so this can be done along other things from within a `lockimport_and_run`-call - while the other locks the importer, gets the block and actually finalizes it. The difference in documentation is subtle but in there.